app/lib/actions/katello/organization/simple_content_access/toggle.rb

```diff
@@ -12,7 +12,8 @@ class Toggle < Actions::AbstractAsyncTask
           attr_reader :organization
 
           def plan(organization_id)
-            @organization = ::Organization.find(organization_id)
+            @organization = ::Organization.find(organization_id.to_i)
+            input[:organization_name] = @organization.name
             action_subject organization
             ::Katello::Resources::Candlepin::Owner.update(@organization.label, contentAccessMode: content_access_mode_value)
```

Comment:
Looking at this whole action chain, there seems to be no run/finalize in any of these. All the work is being done in plan phase which is not ideal. The Actions::AbstractAsyncTask also allows skipping failed tasks in run phase but not in plan. Looking at the code the PrepareContentOverrides does a lot of heavylifting including creating content overrides on all hosts and activation keys in candlepin. We should move that to the run phase. And this final Owner.update to SCA can move to the finalize once the run phase is completed. Thoughts?

Reply:
Sounds like a good ... plan! :smile:

Reply:
Yeah recommend moving it to run (since it involves candlepin.)

Reply:
Moved everything to `run`. I also kept the `Owner.update` in `run`, since it's Candlepin, is that ok?
![image](https://user-images.githubusercontent.com/22042343/233699743-7388d826-6d6a-464e-adeb-17554d08687a.png)